zenodo/modules/deposit/api.py

```diff
@@ -484,8 +487,14 @@ def validate_publish(self):
                 raise MissingCommunityError(missing)
 
     @mark_as_action
-    def publish(self, pid=None, id_=None, user_id=None, sip_agent=None):
+    def publish(self, pid=None, id_=None, user_id=None, sip_agent=None,
+                check_for_spam=True):
         """Publish the Zenodo deposit."""
+        task = check_metadata_for_spam.delay(self['recid'], str(self.id))
+        is_spam = json.loads(task.get())
+        if is_spam and check_for_spam:
+            if not Permission(ActionNeed('admin-access')).can():
+                current_app.config['ZENODO_SPAM_HANDLING_ACTIONS'](self)
```

Comment:
I would wrap this in try/catch with a warning raise in the `catch` block. In case something goes bad in any part of the check, we don't want to completely block users from publishing records.